src/blockchain/node/filteroracle/BlockIndexer.hpp

```diff
@@ -122,11 +122,17 @@ class FilterOracle::BlockIndexer : public BlockDMFilter,
         const UnallocatedCString& shutdown,
         const NotifyCallback& notify) noexcept;
 
-    ~BlockIndexer();
+    ~BlockIndexer() final;
+
+protected:
+    auto pipeline(zmq::Message&& in) noexcept -> void final;
```

Comment:
What is the added value of changing definition of this pipeline ? performance ?

Reply:
No. Originally, some of the subclasses had Message passed as rvalue reference, others as const reference, with the static downcast silently adjusting the call. With a virtual pipeline that now avoids the downcast, all the overrides have to have the same signature.